templates/footer_social_media.twig

```diff
@@ -1,36 +1,40 @@
-<ul class="footer-social-media list-unstyled col-md-8 col-lg-5">
-  {% if social_overrides is defined and social_overrides|length %}
-    {% for social in social_overrides %}
-      <li>
-        <a href="{{ social.url }}"
+{% set total_social_icons = social_overrides is defined and social_overrides|length ? social_overrides|length : footer_social_menu|length %}
+
+<div class="site-footer__social-media site-footer__social-media--{{ nav_type }}">
```

Comment:
I don't think BEM plays very well here.

Reply:
I mean, there's `site-footer` on everything :smile:

Reply:
I don't agree with this because this is how BEM works (block__element--modifier).
`site-footer` is the block, then, every element inside the block should be named as `block__element`.

`site-footer__social-media`: In this case, `site-footer` is the block and `social_media` is the element.
`site-footer__social-media--{{ nav_type }}`: In this case `site-footer` is the block, `social_media` is the element and `--{{ nav_type }}` is the modifier.

So, if you use BEM, `site-footer` will be added in all classnames since is the `block`.

Look at [this](https://simple-web.dev/bem-by-example) example.
> It's important to combine the original class, and the modifier class even if it looks a bit ugly.

Reply:
I understand what the logic behind BEM is, my point is that it doesn't work well in this instance. I'm open to being convinced to using a methodology for naming if you can provide a good argument for it. Just saying 
> I don't agree with this because this is how BEM works 

is not very convincing.

Reply:
Let me go into detail about some things that are not playing well here:
- `site-footer__social-media-list--justify-space-between`: "modifier" in BEM is very open to interpretation and ends up containing all kinds of things. In this case it corresponds to a CSS rule that we happen to put on lists that have 4 to 6 items.
- This uses a `span` with a title, and then uses BEM to make it "semantic". Whereas using an `h2` or other heading would be more semantic, and probably remove the need to add a separate selector, as we could write CSS as `.site-footer__social-media h2`.
- `site-footer__social-media--{{ nav_type }}`: the footer type is already present in [on the `footer` element this is inside](https://github.com/greenpeace/planet4-master-theme/pull/1466/files#diff-9ccc5c1e49803dab74b7d5fb3b391bb54c4521688a0f01b887e1d082a4759e60R1). So we can also write the following CSS `site-footer--minimal .social-media {...}`

Reply:
Although BEM is a nice standard for css coding, we are not using it anywhere else in our codebase (at least not intentionally). So I'd be very careful of introducing it now, since we also need to see how Design Tokens may affect the way we name css classes.

So although `site-footer__social-media-list--justify-space-between` would probably be correct in this case, if we wanted to follow BEM, it strikes to me as completely different compared to the rest of our codebase. My suggestion would be to keep it simple for now, and try to keep the same css classes logic as the current footer. We can re-visit usage of BEM, once we start working on Design Tokens and css variables more extensively.

Reply:
Sorry, my last message was added at the same time as yours @comzeradd.

Thanks both for your comments 😄! I'll remove BEM on my code and hope we should find another way for styling our blocks, maybe `styled-components` 🤞 It could be a good experiment to do 😉

Reply:
I've already updated the PR. Removed all related to BEM. Thanks guys!